File: linked_lists/remove_nth_node_from_list_end.rb
def remove_nth_from_end(head, n)
    pre = ListNode.new
    pre.next, fast, slow = head, pre, pre
    n.times.each { fast = fast.next }

    slow, fast = slow.next, fast.next until fast.next.nil?
    slow.next = slow.next.next
    pre.next
end

File: linked_lists/remove_nth_node_from_list_end.py
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        dummy = ListNode(0)
        dummy.next, fast, slow = head, dummy, dummy
        
        for _ in range(n):
            fast = fast.next

        while fast.next:
            slow = slow.next
            fast = fast.next

        slow.next = slow.next.next
        return dummy.next